test/jdk/tools/jpackage/helpers/jdk/jpackage/test/PackageTest.java

```diff
@@ -317,6 +317,11 @@ public PackageTest configureHelloApp(String javaAppDesc) {
         return this;
     }
 
+    public PackageTest addLauncherName(String name) {
```

Comment:
There is no need to make this function public and explicitly call it from tests.
It would be better to add call to this function from AdditionalLauncher.applyTo(PackageTest).

Reply:
But test calls AdditionalLauncher.applyTo(JPackageCommand cmd), which does not have reference to PackageTest. Do you know why we need two applyTo()?

Reply:
Just change the test code to call AdditionalLauncher.applyTo(PackageTest):
`launcher1.applyTo(packageTest);
launcher2.applyTo(packageTest)`

AdditionalLauncher.applyTo(JPackageCommand) is for non-packaging tests.

Reply:
Actually it will not work. I am running appImageCmd command before test executes to generate app image. If I add it to packageTest, jpackage will fail when trying to run command with app-image and launchers.

Reply:
I will keep addLauncherName() inside test to add launchers manually and will also call it in applyTo(PackageTest) for future tests, that might use such functionality.

Reply:
Can you have applyTo() called for `appImageCmd` and for `packageTest`?